readme.md

```diff
@@ -115,81 +115,184 @@ Case classes are serialized using the `apply` and `unapply` methods on their com
 Anything else is not supported by default, but you can add support using Custom Picklers.
 
 ## Defaults
-If a field is missing upon deserialization, weePickle uses the default value if one exists.
+If a field is missing upon **deserialization**, weePickle uses the default value if one exists:
 
 ```scala
 case class Dflt(i: Int = 42)
+object Dflt {
+  implicit val rw: FromTo[Dflt] = macroFromTo
+}
 
-FromJson("""{}""").transform(ToScala[Dflt])          ==> Dflt(42)
-FromJson("""{"i": 999}""").transform(ToScala[Dflt])  ==> Dflt(999)
+FromJson("""{}""").transform(ToScala[Dflt])            ==> Dflt(42)
+FromJson("""{"i": null}""").transform(ToScala[Dflt])   ==> Dflt(42)
+FromJson("""{"i": 999}""").transform(ToScala[Dflt])    ==> Dflt(999)
 ```
 
-If a field at serialization time has the same value as the default, it will be written unless annotated with `@dropDefault`.
+If a field at **serialization** time has the same value as the default, it will be written unless annotated with `@dropDefault`:
 
 ```scala
-FromScala(Dflt(42)).transform(ToJson.string)         ==> """{"i": 42}"""
+FromScala(Dflt(42)).transform(ToJson.string)    ==> """{"i": 42}"""
+FromScala(Dflt(999)).transform(ToJson.string)   ==> """{"i": 999}"""
 ```
 
+### using `@dropDefault`
+
+When **serializing**, any field annotated `@dropDefault` and a value equal to their default will be dropped from the JSON:
+
 ```scala
-case class Dflt2(@dropDefault i: Int = 42)
-FromScala(Dflt2(42)).transform(ToJson.string)        ==> """{}"""
+case class DropDflt(@dropDefault i: Int = 42)
+object DropDflt {
+  implicit val rw: FromTo[DropDflt] = macroFromTo
+}
+
+FromScala(DropDflt(42)).transform(ToJson.string)    ==> """{}"""
+FromScala(DropDflt(999)).transform(ToJson.string)   ==> """{"i": 999}"""
 ```
 
-If a class is annotated with `@dropDefault`, all fields with default values will not be written.
+If a ***class*** is annotated with `@dropDefault`, all fields with the same value as their default will not be written:
 
 ```scala
-@dropDefault case class Dflt3(i: Int = 42, j: Int = 43, k: Int = 45)
-FromScala(Dflt2(42, 43, 0)).transform(ToJson.string)        ==> """{"k": 0}"""
+@dropDefault case class DropMultiDflts(i: Int = 42, j: Int = 43, k: Int = 44)
+object DropMultiDflts {
+  implicit val rw: FromTo[DropMultiDflts] = macroFromTo
+}
+
+FromScala(DropMultiDflts(i = 42, j = 43, k = 999)).transform(ToJson.string)   ==> """{"k": 999}"""
 ```
 
+If attempting to deserialize JSON to a field that has no defaults, and the JSON value is invalid (e.g. missing, or `null`), the operation will throw an Abort Exception.
```

Comment:
Mostly correct. `null` handling is up to the `To`. Most will throw, but [Option](https://github.com/rallyhealth/weePickle/blob/v1/weepickle-implicits/src/main/scala/com/rallyhealth/weepickle/v1/implicits/DefaultTos.scala#L234), [Array](https://github.com/rallyhealth/weePickle/blob/v1/weepickle-implicits/src/main/scala/com/rallyhealth/weepickle/v1/implicits/DefaultTos.scala#L248), and others may override `visitNull`. The case class macro itself doesn't do any special null-handling.